assets/js/modules/optimize/datastore/settings.js

```diff
@@ -16,20 +16,32 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
 /**
  * Internal dependencies
  */
 import API from 'googlesitekit-api';
 import Data from 'googlesitekit-data';
 import { TYPE_MODULES } from '../../../components/data/constants';
 import { invalidateCacheGroup } from '../../../components/data/invalidate-cache-group';
+import { createStrictSelect, createValidationSelector } from '../../../googlesitekit/data/utils';
 import {
 	isValidOptimizeID,
 	isValidAMPExperimentJSON,
 } from '../util';
 import { STORE_NAME } from './constants';
 
-const { createRegistrySelector, createRegistryControl } = Data;
+const { createRegistryControl } = Data;
+
+// Invariant error messages.
+export const INVARIANT_DOING_SUBMIT_CHANGES = 'cannot submit changes while submitting changes';
+export const INVARIANT_DONT_HAVE_SETTINGS_CHANGED = 'cannot submit changes if settings have not changed';
+export const INVARIANT_INVALID_AMP_EXPIREMENT_JSON = 'require an ampExperimentJSON to be valid JSON if set';
```

Comment:
Typo fix and slight rewording.
```suggestion
export const INVARIANT_INVALID_AMP_EXPERIMENT_JSON = 'ampExperimentJSON must be valid JSON if set';
```